packages/api/src/external/carequality/document/process-ihe-to-gw-document-retrieval.ts

```diff
@@ -2,50 +2,73 @@ import { out } from "@metriport/core/util/log";
 import { Bundle, BundleEntry } from "@medplum/fhirtypes";
 import { errorToString } from "@metriport/core/util/error/shared";
 import { capture } from "@metriport/core/util/notifications";
-import { DocumentReference } from "@metriport/ihe-gateway-sdk";
+import { DocumentReference, DocumentRetrievalRespFromExternalGW } from "@metriport/ihe-gateway-sdk";
 import { MedicalDataSource } from "@metriport/core/external/index";
-import { DocumentRetrievalResult } from "../document-retrieval-result";
 import { isConvertible } from "../../fhir-converter/converter";
-import { setDocQueryProgressWithSource } from "../../hie/set-doc-query-progress-with-source";
-import { tallyDocQueryProgressWithSource } from "../../hie/tally-doc-query-progress-with-source";
+import { setDocQueryProgress } from "../../hie/set-doc-query-progress";
+import { tallyDocQueryProgress } from "../../hie/tally-doc-query-progress";
 import { convertCDAToFHIR } from "../../fhir-converter/converter";
 import { upsertDocumentsToFHIRServer } from "../../fhir/document/save-document-reference";
-import { cqToFHIR } from "../../fhir/document";
+import { cqToFHIR } from "./shared";
 import { getDocuments } from "../../fhir/document/get-documents";
 import { ingestIntoSearchEngine } from "../../aws/opensearch";
 
-export async function processDocumentRetrievalResults({
+export async function processIHEToExternalGwDocumentRetrievals({
   requestId,
   patientId,
   cxId,
-  documentRetrievalResults,
+  resultsOfAllExternalGWs,
 }: {
   requestId: string;
   patientId: string;
   cxId: string;
-  documentRetrievalResults: DocumentRetrievalResult[];
+  resultsOfAllExternalGWs: DocumentRetrievalRespFromExternalGW[];
 }): Promise<void> {
   try {
-    for (const result of documentRetrievalResults) {
-      const { operationOutcome } = result.data;
-      const issuesWithGateway = operationOutcome?.issue?.length ?? 0;
-      const successDocsCount = result.data.documentReference?.length ?? 0;
-
-      tallyDocQueryProgressWithSource({
-        patient: { id: patientId, cxId: cxId },
-        progress: {
-          successful: successDocsCount,
-          errors: issuesWithGateway,
-        },
-        type: "download",
-        requestId,
-        source: MedicalDataSource.CAREQUALITY,
-      });
+    let issuesWithGateway = 0;
+    let successDocsCount = 0;
 
-      await handleDocReferences(result.data.documentReference, requestId, patientId, cxId);
+    for (const resultOfExternalGW of resultsOfAllExternalGWs) {
```

Comment:
nit: result > response ? The result is the outcome of this function, from the use case perspective